test/core/tsi/ssl_transport_security_utils_test.cc

```diff
@@ -429,6 +445,184 @@ INSTANTIATE_TEST_SUITE_P(FrameProtectorUtil, FlowTest,
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kIntermediateCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), 1);
+}
+
+TEST(CrlUtils, VerifySignatureModifiedSignature) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kModifiedSignature, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), 0);
+}
+
+TEST(CrlUtils, VerifySignatureModifiedContent) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kModifiedContent, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  EXPECT_EQ(crl, nullptr);
+}
+
+TEST(CrlUtils, VerifySignatureWrongIssuer) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kValidCrl, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kIntermediateCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), 0);
+}
+
+TEST(CrlUtils, VerifySignatureWrongIssuer2) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kIntermediateCrl, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), 0);
+}
+
+TEST(CrlUtils, VerifySignatureNullCrl) {
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = nullptr;
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), -1);
+}
+
+TEST(CrlUtils, VerifySignatureNullCert) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kIntermediateCrl, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = nullptr;
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), -1);
+}
+
+TEST(CrlUtils, VerifySignatureNullCrlAndCert) {
```

Comment:
I think this test only really catches the "is crl null" case and we need a separate test for the "is issuer null" case.

Reply:
I have a test for just null CRL and just null cert above this, figured I'd have the complete set even if the test doesn't do much else